tfhe/src/integer/server_key/radix_parallel/tests_cases_unsigned.rs

```diff
@@ -0,0 +1,2992 @@
+    }
+}
+
+pub(crate) fn unchecked_scalar_mul_corner_cases_test<P, T>(param: P, mut executor: T)
+where
+    P: Into<PBSParameters>,
+    T: for<'a> FunctionExecutor<(&'a RadixCiphertext, u64), RadixCiphertext>,
+{
+    let (cks, sks) = KEY_CACHE.get_from_params(param);
+    let nb_ct =
+        (128f64 / (cks.parameters().message_modulus().0 as f64).log2().ceil()).ceil() as usize;
+    let cks = RadixClientKey::from((cks, nb_ct));
+
+    executor.setup(&cks, sks.clone());
+
+    // This example will not pass if the terms reduction is wrong
+    // on the chunk size it uses to reduce the 'terms' resulting
+    // from blockmuls
+    {
+        let clear = 307096569525960547621731375222677666984u128;
+        let scalar = 5207034748027904122u64;
+
+        let ct = cks.encrypt(clear);
+        let ct_res = executor.execute((&ct, scalar));
+        let dec_res: u128 = cks.decrypt(&ct_res);
+        assert_eq!(clear.wrapping_mul(scalar as u128), dec_res);
+
+        let clear = u128::MAX;
+        let scalar = u64::MAX;
+
+        let ct = cks.encrypt(clear);
+        let ct_res = executor.execute((&ct, scalar));
+        let dec_res: u128 = cks.decrypt(&ct_res);
+        assert_eq!(clear.wrapping_mul(scalar as u128), dec_res);
+    }
+
+    // Trying to multiply a ciphertext with a scalar value
+    // bigger than the ciphertext modulus should work
+    {
+        let cks: crate::integer::ClientKey = cks.into();
+        let nb_ct =
+            (8f64 / (cks.parameters().message_modulus().0 as f64).log2().ceil()).ceil() as usize;
+        let cks = RadixClientKey::from((cks, nb_ct));
+
+        executor.setup(&cks, sks);
+
+        let clear = 123u64;
+        let scalar = 17823812983255694336u64;
+        assert_eq!(scalar % 256, 0);
+
+        let ct = cks.encrypt(clear);
+        let ct_res = executor.execute((&ct, scalar));
+        let dec_res: u64 = cks.decrypt(&ct_res);
+        assert_eq!(clear.wrapping_mul(scalar) % 256, dec_res);
+    }
+}
+
+pub(crate) fn unchecked_scalar_left_shift_test<P, T>(param: P, mut executor: T)
+where
+    P: Into<PBSParameters>,
+    T: for<'a> FunctionExecutor<(&'a RadixCiphertext, u64), RadixCiphertext>,
+{
+    let (cks, sks) = KEY_CACHE.get_from_params(param);
+    let cks = RadixClientKey::from((cks, NB_CTXT));
+
+    let mut rng = rand::thread_rng();
+
+    // message_modulus^vec_length
+    let modulus = cks.parameters().message_modulus().0.pow(NB_CTXT as u32) as u64;
+    let nb_bits = modulus.ilog2();
+
+    executor.setup(&cks, sks);
+
+    for _ in 0..NB_TEST {
+        let clear = rng.gen::<u64>() % modulus;
+        let scalar = rng.gen::<u32>();
+
+        let ct = cks.encrypt(clear);
+
+        // case when 0<= scalar < nb_bits
+        {
+            let scalar = scalar % nb_bits;
+            let encrypted_result = executor.execute((&ct, scalar as u64));
+            let decrypted_result: u64 = cks.decrypt(&encrypted_result);
+            let expected = (clear << u64::from(scalar)) % modulus;
```

Comment:
any reason for the need of u64::from ?

also << vs wrapping_shl ?

Reply:
wrapping_shl is useless here as wrapping_shl is equivalent to `shift << Type::BITS`n and here we use a u64 type
but our values are meant to mimic a type with much less bits